tensordict/tensordict.py

```diff
@@ -1617,9 +1617,28 @@ def flatten_keys(
         self, separator: str = ".", inplace: bool = False
     ) -> TensorDictBase:
         to_flatten = []
+        already_flat = []
         for key, meta_value in self.items_meta():
             if meta_value.is_tensordict():
                 to_flatten.append(key)
+            else:
+                already_flat.append(key)
+
+        def exists(obj, chain):
+            _key = chain.pop(0)
+            if isinstance(obj, Tensor):
+                return None
+            if _key in obj:
+                return exists(obj[_key], chain) if chain else obj[_key]
+
+        for key in already_flat:
+            if separator in key:
+                _existing_val = exists(self.to_dict(), key.split(separator))
+
+                if isinstance(_existing_val, Tensor):
```

Comment:
Careful here: a memmap-tensor is a tensor like (it behaves like a tensor) but it does not inherit from it.
I'd rather say `if not isinstance(_existing_val, TensorDictBase)`